### PR TITLE
[MANUAL MIRROR] Reworks Duffel Bags (Zippers)

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Manual mirror of https://github.com/tgstation/tgstation/pull/76313 and https://github.com/tgstation/tgstation/pull/76442

Closes #22136, #22174

## How This Contributes To The Skyrat Roleplay Experience
Upstream mirror good :)

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

I will post photos of things working soon.

</details>

## Changelog

:cl:
balance: Duffelbags will now only make you slow while they are unzipped. As a tradeoff, you now need to stand still and zip/unzip them to access their contents/not move real slow.
 fix: Fixes some fuck with duffelbags, them not holding enough + issues with spawning gear in them (job shit and all)
/:cl:

### CODE REVIEW

File: code/modules/antagonists/cult/cult.dm [hunk 1]
```diff
@@ -50,6 +50,54 @@
...
+	var/where = mob.equip_in_one_of_slots(T, slots, indirect_action = TRUE)
+	if(!where)
+		to_chat(mob, span_userdanger("Unfortunately, you weren't able to get a [item_name]. This is very bad and you should adminhelp immediately (press F1)."))
+		return FALSE
+	else
+		to_chat(mob, span_danger("You have a [item_name] in your [where]."))
+		if(where == "backpack")
+			mob.back.atom_storage?.show_contents(mob)
+		return TRUE
+
```
Comment (softcerv) [inline code/modules/antagonists/cult/cult.dm]:
```suggestion
```

**Review by lessthnthree — APPROVED** [review]

File: code/game/objects/items/storage/backpack.dm [hunk 2]
```diff
@@ -398,12 +389,94 @@
...
 	icon_state = "duffel"
 	inhand_icon_state = "duffel"
+	actions_types = list(/datum/action/item_action/zipper)
+	storage_type = /datum/storage/duffel
+	var/zip_slowdown = 1
+	/// If this bag is zipped (contents hidden) up or not
+	/// Starts enabled so people need to figure it out to use the thing
+	var/zipped_up = TRUE
 	//slowdown = 1 //ORIGINAL
 	slowdown = 0.5 //SKYRAT EDIT CHANGE
```
Comment (Nerev4r) [inline code/game/objects/items/storage/backpack.dm]: I don't think this is supposed to still be here? Unless slowdown doesn't impact zip_slowdown?

**Comment by W4rd3nn** [comment]
Is there still supposed to be slow down while they're zipped up? You're ever so slightly slower with it zipped up than you are not wearing it at all

**Comment by Nerev4r** [comment]
> Is there still supposed to be slow down while they're zipped up? You're ever so slightly slower with it zipped up than you are not wearing it at all

Probably the result of what I posted earlier.

**Comment by softcerv** [comment]
> Is there still supposed to be slow down while they're zipped up? You're ever so slightly slower with it zipped up than you are not wearing it at all

I forgor to remove it.

**Comment by softcerv** [comment]
Golden said it was fine for me to do this

**Pull request merged**